wsd/DocumentBroker.cpp

```diff
@@ -702,6 +704,9 @@ bool DocumentBroker::download(const std::shared_ptr<ClientSession>& session, con
         wopiInfo->set("SupportsRename", wopifileinfo->getSupportsRename());
         wopiInfo->set("UserCanRename", wopifileinfo->getUserCanRename());
         wopiInfo->set("FileUrl", wopifileinfo->getFileUrl());
+        wopiInfo->set("IsFreemiumUser", wopifileinfo->isFreemiumUser());
+        wopiInfo->set("FreemiumBlacklist", Freemium::FreemiumManager::getFreemiumBlacklist());
```

Comment:
Why? What has WOPI to do with any of this?

Reply:
we get the Freemium user info from richdocuments so it is initialized in wopi.

there are just 2 info we need to worry about, blocked command and if user is freemium, to me it makes sense to send both info together to leaflet rather than separately because without this info we can not initialized the UI. This is the best place to have for sure.

Reply:
> we get the Freemium user info from richdocuments so it is initialized in wopi.

I don't follow the logic. Freemium isn't user-specific, so I'm not sure how it's related to users at all. Even less related to WOPI, which is the document storage.

What are we going to do if the document came from non-wopi storage?

> there are just 2 info we need to worry about, blocked command and if user is freemium, to me it makes sense to send both info together to leaflet rather than separately because without this info we can not initialized the UI. This is the best place to have for sure.

I don't see why we can't send them back in a different message together. The wopiinfo isn't the only place we can have two things together.

Reply:
Freemium is user specific

Reply:
It was suggested to me to push freemium status via file info so that's why it is this way. If it's non wopi than it does not work. These freemium feature is for providers so i don't think there would be case where wopi isn't available. Or at least for now we only have support for wopi i think

Reply:
done